### ISSUE TITLE
Show matches count when searching

### ISSUE THREAD

**Issue by XyLoNaMiyX** [notepadqq/notepadqq#304]
Hello! First I want to thank you for the program, it's amazing :)

A small addition I'd love to see is displaying the "matches count" when searching something from the _Search dialog_. Or even a _Count_ button would be nice!

**Comment by JuBan1** [comment]
Yes, that's sorely lacking. A Count button shouldn't be much work- I'll see what I can do.

On a side note- your first name doesn't happen to be Imanol, does it?

**Comment by XyLoNaMiyX** [comment]
> On a side note- [...]

Well, you guessed the first name, but not the R. Now edit these comments so nobody in the internet finds out! I try to keep my identity hidden and you just spoiled it!! 😡😒

**Comment by JuBan1** [comment]
Sorry, wasn't my intent. I knew a person with the same handle and first name who also played games and, if I recall correctly, was also from the same country. So I thought it's a long shot but I might as well ask.

**Comment by XyLoNaMiyX** [comment]
No worries, I was only joking ✌. Email me if you want to talk about that (not to fill this issue with comments).

**Comment by danieleds** [comment]
I just wanted to point out that if you use the "Select All" button, you can find the total number of matches in the status bar:

![search](https://cloud.githubusercontent.com/assets/402652/19838862/0cd21020-9ed8-11e6-8436-d2c7c9c0096f.png)

(well, I agree I wouldn't call that intuitive)

**Comment by XyLoNaMiyX** [comment]
I tried looking there @danieleds! But I didn't really know on what number I should focus on, and probably missed the _Select all_ part too 😁

**Comment by JuBan1** [comment]
That's a bit of a cheat, though. The number in parenthesis shows the number of lines currently selected. It just happens to coincide with the number of search results if Select All is clicked. But that stops working if the search regex contains newlines.

**Comment by danieleds** [comment]
It is the number of selections, not the number of lines.

**Comment by JuBan1** [comment]
Does it have different meanings depending on whether one or multiple selections exist? Because when having a single piece of text selected it definitely counts the number of lines.

![image](http://i.imgur.com/6yGNKH2.png)

**Comment by danieleds** [comment]
Sorry my mistake, you're right, that number is not what was asked :)

(The actual meaning of that number is different and indicates how many "placeholders" are needed to paste the text somewhere else. And each selected line is counted once, but all different selections within the same line contribute to the counter.)

**Comment by JuBan1** [comment]
> I was thinking if maybe there's a more "natural" way to display the count instead of a button.

We could have a label in the search dialog that updates automatically but I shied away from this because the counting would have to be re-done on every keystroke (when inputting a search term). On larger docs that might be too much. I don't know for sure though.

**Comment by XyLoNaMiyX** [comment]
> [...] the counting would have to be re-done [...]

Actually, the search is done on every keystroke (which I personally find kind of cool, it gives it a smooth feel), scrolling to the where the match was found. So I doubt this would be an issue, yet I don't know how it's handled in large documents (is it ran in a different thread? Can the search be canceled after another keystroke? etc.)

**Comment by danieleds** [comment]
> but I shied away from this because the counting would have to be re-done on every keystroke (when inputting a search term). On larger docs that might make searching laggy. I don't know for sure though.

Yeah I agree, that's what I fear too. A lot of editors do that, and I think it's a great ux, but we should see how it performs in combination with our advanced search.

**Comment by XyLoNaMiyX** [comment]
Settings > Preferences > Search > Search as I type.

I think auto-count should be enabled when that is enabled. Perhaps you could add a check box and a numeric field that said _[x] Only if the file is smaller than... [10] MB_ (for example).
